cluster-autoscaler/core/scale_down.go

```diff
@@ -55,6 +55,9 @@ const (
 	ScaleDownNodeDeleted ScaleDownResult = iota
 	// ScaleDownDisabledKey is the name of annotation marking node as not eligible for scale down.
 	ScaleDownDisabledKey = "cluster-autoscaler.kubernetes.io/scale-down-disabled"
+	// ScaleDownNonEmptyCandidatesCount is the maximum number of non empty nodes
+	// considered at once as candidates for scale down.
+	ScaleDownNonEmptyCandidatesCount = 30
```

Comment:
Make it a flag.

Reply:
We already have a ton of arcane flags. Even as CA developer I generally have to look into the code to understand what they do and how to use them. And we never really test anything with their values changed.

Bottom line is I don't think anyone will ever change this and I think we should be aiming at reducing the numbers of arcane flags controlling some internal heuristics, not adding new ones.

Reply:
I'm with Maciek on this one. However I name this flag and however long the description, it's going to be really hard to figure out what value to set. The only thing that kinda makes sense is what Maciek mentioned offline - add a flag to switch this off completely. This gives at least a somewhat clearer choice: better performance/responsiveness vs better scaling behavior

Reply:
The thing is how fast can we react to any problems that are caused with the numbers picked in a wrong way. In GKE we can react much quicker if the configuration option is exposed as a flag.

Reply:
Made this a flag.